internal/beatcmd/beat.go

```diff
@@ -266,6 +267,20 @@ func (b *Beat) Run(ctx context.Context) error {
 		)
 	}
 
+	if runtime.GOOS == "darwin" {
+		v, err := kernelVersion()
+		if err != nil {
+			logger.Warnf("failed to retrieve kernel version, ignoring potential deprecation warning: %v", err)
+		}
```

Comment:
Should we use `go-sysinfo`?

```go
if host, err := sysinfo.Host(); err != nil {
	logger.Warnf("failed to retrieve kernel version, ignoring potential deprecation warning: %v", err)
} else if host.Info().OS.Major == 19 {
	// macOS 10.15.x (catalina) means darwin kernel 19.y
	...
}
```

Then there's no need for `kernel.go`. Generally I'd prefer to keep platform-specific code out of apm-server, and isolated in libraries like go-sysinfo.

Reply:
I've looked at the go-sysinfo implementation and it does not rely on CGO. However it is reading from a file (https://github.com/elastic/go-sysinfo/blob/main/providers/darwin/os.go#L39), is that a concern ? 

___

The other issue I'm having is with the results: I'm testing the changes on github with the following action:

```
name: GitHub Actions Demo
on: [push]
jobs:
  build:
    runs-on: macos-12
    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-go@v3.5.0
        with:
          go-version: 1.19
      - name: Run
        run: CGO_ENABLED=0 go run main.go
```

The syscall version is returning `21.6.0` which is mapped to macOS 12.5 (consistently with the github action target).

On the other hand, go-sysinfo returns:
```
&{macos darwin darwin Mac OS X 10.16 10 16 0 21G419 }
Major: 10
Minor: 16
```

That's unexpected. Am I missing something ?

Reply:
UPDATE:

I looked at the release page (https://opensource.apple.com/releases/): macOS 10.16 doesn't even exist :cold_sweat: 

`host.Info().KernelVersion` seems to be consistent with the custom syscall and the version specified in the github action. It does not rely on CGO so it should be fine (https://github.com/elastic/go-sysinfo/blob/main/providers/darwin/kernel_darwin.go)

Reply:
https://eclecticlight.co/2020/08/13/macos-version-numbering-isnt-so-simple/
weird :)

Anyway, using `host.Info().KernelVersion` looks good.